packages/integration_test/ios/Classes/FLTIntegrationTestRunner.m

```diff
@@ -0,0 +1,60 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTIntegrationTestRunner.h"
+
+#import "IntegrationTestPlugin.h"
+
+@import ObjectiveC.runtime;
+@import UIKit;
+
+@interface FLTIntegrationTestRunner ()
+
+@property IntegrationTestPlugin *integrationTestPlugin;
+
+@end
+
+@implementation FLTIntegrationTestRunner
+
+- (instancetype)init {
+  self = [super init];
+  _integrationTestPlugin = [IntegrationTestPlugin instance];
+
+  return self;
+}
+
+- (void)testIntegrationTestWithResults:(NS_NOESCAPE FLTIntegrationTestResults)testResult {
+  IntegrationTestPlugin *integrationTestPlugin = self.integrationTestPlugin;
+  UIViewController *rootViewController = UIApplication.sharedApplication.delegate.window.rootViewController;
+  if (![rootViewController isKindOfClass:[FlutterViewController class]]) {
+    testResult(NSSelectorFromString(@"testSetup"), NO, @"rootViewController was not expected FlutterViewController");
+  }
+  FlutterViewController *flutterViewController = (FlutterViewController *)rootViewController;
+  [integrationTestPlugin setupChannels:flutterViewController.engine.binaryMessenger];
+
+  // Spin the runloop.
+  while (!integrationTestPlugin.testResults) {
+    [NSRunLoop.currentRunLoop runUntilDate:[NSDate dateWithTimeIntervalSinceNow:1.0]];
+  }
+
+  [integrationTestPlugin.testResults enumerateKeysAndObjectsUsingBlock:^(NSString *test, NSString *result, BOOL *stop) {
+    // Create an appropriate XCTest method name based on the dart test name.
+    // Example: dart test "verify widget" becomes "testVerifyWidget"
+    NSString *upperCamelTestName = [test.localizedCapitalizedString stringByReplacingOccurrencesOfString:@" " withString:@""];
```

Comment:
I missed this last time, but it seems like we need a lot more normalization here, since AFAIK a Dart test name can be an arbitrary string. E.g., If the test name is `"validate multi-point unicode: 😁 "` we're going to have a bad time.

Reply:
This is probably going to be complicated enough that we should pull it out into a helper that has its own set of unit tests.

Reply:
Alternately you could sanitize unicode to a byte description (e.g., 🌈 -> "U1F308") to avoid collisions, but it's a pretty unlikely edge case.

Reply:
Let's see if anyone ever files an issue about it...